ground/src/main/java/com/google/android/ground/ui/map/gms/mog/MogTileDownloader.kt

```diff
@@ -27,19 +27,22 @@ import java.io.File
  * @param outputBasePath the base path on the local file system where tiles should be written.
  */
 class MogTileDownloader(private val client: MogClient, private val outputBasePath: String) {
+  // TODO(#1755): Add test coverage to this class and entire package.
   /**
    * Executes the provided [requests], writing resulting tiles to [outputBasePath] in sub-paths of
    * the form `{z}/{x}/{y}.jpg`.
    */
-  @RequiresApi(Build.VERSION_CODES.S)
-  suspend fun downloadTiles(requests: List<MogTilesRequest>) {
-    // TODO: Return download status as Flow.
+  suspend fun downloadTiles(requests: List<MogTilesRequest>) = flow {
     client.getTiles(requests).collect { tile ->
       val (x, y, zoom) = tile.metadata.tileCoordinates
       val path = File(outputBasePath, "$zoom/$x")
       path.mkdirs()
       val gmsTile = tile.toGmsTile()
-      File(path, "$y.jpg").writeBytes(gmsTile.data!!)
+      val outFile = File(path, "$y.jpg")
+      val data = gmsTile.data!!
+      outFile.writeBytes(data)
+      Timber.d("Saved ${data.size} bytes to ${outFile.path}")
+      emit(data.size)
```

Comment:
Should we emit the path of the downloaded tile as well? Is the downloaded size being used at the caller place?